src/main/java/us/kbase/userprofile/MongoController.java

```diff
@@ -53,77 +52,72 @@ public MongoController(final String host, final String database) {
 	}
 	
 	public MongoController(final String host, final String database,
-			final String mongoUser, final String mongoPswd) {
-		final DB db = getDB(host, database, mongoUser, mongoPswd);
+						   final String mongoUser, final String mongoPswd) {
+		final MongoDatabase db = getDB(host, database, mongoUser, mongoPswd);
 		profiles = db.getCollection(COL_PROFILES);
 		//jProfiles = jongo.getCollection(COL_PROFILES);
 		ensureIndex();
 	}
 	
-	private DB getDB(final String host, final String db, final String user, final String pwd) {
+	private MongoDatabase getDB(final String host, final String db, final String user, final String pwd) {
 		// TODO update to non-deprecated APIs
+		final MongoClientSettings.Builder mongoBuilder = MongoClientSettings.builder().applyToClusterSettings(
+				builder -> builder.hosts(Arrays.asList(new ServerAddress(host))));
 		final MongoClient cli;
 		if (user != null) {
 			final MongoCredential creds = MongoCredential.createCredential(
 					user, db, pwd.toCharArray());
 			// unclear if and when it's safe to clear the password
-			cli = new MongoClient(new ServerAddress(host), creds,
-					MongoClientOptions.builder().build());
+			cli = MongoClients.create(mongoBuilder.credential(creds).build());
 		} else {
-			cli = new MongoClient(new ServerAddress(host));
+			cli = MongoClients.create(mongoBuilder.build());
 		}
-		return cli.getDB(db);
+		return cli.getDatabase(db);
 	}
-	
-	
+
+
 	public List<User> filterUsers(String filter) {
+		List<User> users = new ArrayList<>();
 		
 		if(filter.trim().isEmpty()) {
 			//return all
-			DBCursor cursor = profiles.find(new BasicDBObject(),
-							new BasicDBObject().append("user",1));
-			List<User> users = new ArrayList<User>(cursor.count());
-			while(cursor.hasNext()) {
+			FindIterable<Document> docs = profiles.find(new Document()).projection(new Document("user", 1));
+			for (Document document : docs) {
+				Document d = document.get("user", Document.class);
+				if (d == null || !d.containsKey("username")) continue;
 				User u = new User();
-				DBObject d = (DBObject)cursor.next().get("user");
-				if(!d.containsField("username")) continue;
 				u.setUsername(d.get("username").toString());
-				if(d.containsField("realname")) {
-					if(d.get("realname")!=null) {
-						u.setRealname(d.get("realname").toString());
-					}
+
+				if (d.containsKey("realname") && d.get("realname") != null) {
+					u.setRealname(d.get("realname").toString());
 				}
-				if(d.containsField("thumbnail")) {
-					if(d.get("thumbnail")!=null) {
-						u.setThumbnail(d.get("thumbnail").toString());
-					}
+				if (d.containsKey("thumbnail") && d.get("thumbnail") != null) {
+					u.setThumbnail(d.get("thumbnail").toString());
 				}
-				//System.out.println(u);
```

Comment:
There are numerous "wow"s in the codebase, not going to make a comment for each one

Reply:
Do you want me to delete these print statements. 
There are a bunch of them in the codebase...

Reply:
Hmm, I think I put this comment in the wrong place, and I don't remember where it was supposed to go. 

Up to you re the print statements. If this was wasn't commented it'd be pretty crazy since it could log 40k users on an unfiltered call

Reply:
removed

Reply:
:+1: